src/java.base/share/classes/sun/reflect/generics/reflectiveObjects/TypeVariableImpl.java

```diff
@@ -135,13 +133,9 @@ public Type[] getBounds() {
      * @since 1.5
      */
     public D getGenericDeclaration() {
-        if (genericDeclaration instanceof Class<?> c)
-            ReflectUtil.checkPackageAccess(c);
-        else if ((genericDeclaration instanceof Method) ||
-                (genericDeclaration instanceof Constructor))
-            ReflectUtil.conservativeCheckMemberAccess((Member)genericDeclaration);
-        else
-            throw new AssertionError("Unexpected kind of GenericDeclaration");
+        assert genericDeclaration instanceof Class<?> ||
+                genericDeclaration instanceof Method ||
+                genericDeclaration instanceof Constructor : "Unexpected kind of GenericDeclaration";
```

Comment:
Can remove this; check is already done in `make` factory.

Reply:
I had hoped to keep the change focused on the removal of the left over ReflectUtil.checkPackageAccess, the else throw AssertionError was a surprise to see. So yes, I agree it can be removed but I'd prefer not touch it now, future work in this area can remove it.

Reply:
Indeed, we should clean this up with things like MethodUtil.

Reply:
MethodUtil is for APIs that do reflective stuff on behave of callers so not tied to the SM execution mode. That is for another time.